File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.py
class Solution(object):
    def rearrangeArray(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        res=[0]*len(nums)
        pos_index,neg_index=0,1
        for num in nums:
            if num>0:
                res[pos_index]=num
                pos_index+=2
            else:
                res[neg_index]=num
                neg_index+=2
        return res

File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var rearrangeArray = function(nums) {
    let neg_idx=1;
    let pos_idx=0;
    let res=new Array(nums.length);
    for(let i=0;i<nums.length;i++){
        if(nums[i]>0){
            res[pos_idx]=nums[i];
            pos_idx+=2;
        }
        else{
            res[neg_idx]=nums[i];
            neg_idx+=2;
        }
    }
    return res;
};